x-pack/plugins/profiling/server/lib/setup/is_super_user.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import { ProfilingPluginStartDeps } from '../../types';
+
+export function isSuperuser({
```

Comment:
We do not recommend checking for role names. What specific cluster, index, and/or application privileges are required in order to accomplish your goals?

Reply:
It should be able to:
- Fetch and create new integrations on fleet
- Add some settings to the cluster
- Check some indices status (created or not and its version)

Reply:
Once you've identified the [specific privileges](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html) you require, you can use the security plugin to perform a privileges check:

```ts
const checkPrivileges = plugins.security?.authz.checkPrivilegesDynamicallyWithRequest(request);

const { hasAllRequested } = await checkPrivileges({
  elasticsearch: {
    cluster: [
      /* ... */
    ],
    index: {
      /* ...*/
    },
  },
});

if (hasAllRequested) {
  // User has all requested privileges
} else {
  // User does not have all requested privileges
}